File: surfsense_local/frontend/src/features/model-selection/model-type.ts
/**
 * What a model is for, mirroring the backend's `ModelType`. The user picks one
 * model per type, so a type is also the slot a selection fills.
 */
export const MODEL_TYPES = [
  "text_gen",
  "image_gen",
  "image_edit",
  "video_gen",
  "audio_gen",
] as const

export type ModelType = (typeof MODEL_TYPES)[number]

File: surfsense_local/backend/modules/llm/model_type.py
from enum import StrEnum


class ModelType(StrEnum):
    """What a model is for. Classifiers produce it, and a selection is keyed by it.

    The user picks one model per type, so a type is also a slot. A type no
    feature reads yet can still be chosen; the feature that first reads it
    brings the client that calls it.
    """

    TEXT_GEN = "text_gen"
    IMAGE_GEN = "image_gen"
    IMAGE_EDIT = "image_edit"
    VIDEO_GEN = "video_gen"
    AUDIO_GEN = "audio_gen"
